app/views/state_file/questions/name_dob/edit.html.erb

```diff
@@ -77,6 +91,17 @@
         <%= ff.cfa_input_field(:first_name, t("general.first_name"), classes: ["form-width--long"], options: {disabled: true}) %>
         <%= ff.cfa_input_field(:last_name, t("general.last_name"), classes: ["form-width--long"], options: {disabled: true}) %>
 
+        <% child_birth_yr = current_intake.direct_file_data.eitc_eligible_dependents[ff.object.ssn]&.at("ChildBirthYr")&.content %>
+        <% if child_birth_yr.present? && current_intake.state_code == "ny" %>
+          <div class="form-group">
+            <%= ff.warning_for_date("#{ff.object_name}[dob_year]", nil, nil, child_birth_yr, t(".dependent_dob_year_warning")) %>
+          </div>
+        <% elsif ff.object.ctc_qualifying %>
+          <div class="form-group">
+            <%= ff.warning_for_date("#{ff.object_name}[dob_year]", MultiTenantService.statefile.end_of_current_tax_year.years_ago(17), nil, nil, t(".dependent_dob_under_17_warning")) %>
```

Comment:
Consider extracting `MultiTenantService.statefile.end_of_current_tax_year` so that here and `@sixty_five_years_ago` can both use it and not be inconsistent about extracting a variable for one age and not the other.